tests/models/test_renderedtifields.py

```diff
@@ -172,7 +172,11 @@ def test_delete_old_records(self, rtif_num, num_to_keep, remaining_rtifs, expect
         assert rtif_num == len(result)
 
         # Verify old records are deleted and only 'num_to_keep' records are kept
-        with assert_queries_count(expected_query_count):
+        # For other DBs,an extra query is fired in RenderedTaskInstanceFields.delete_old_records
+        expected_query_count_based_on_db = (expected_query_count + 1, expected_query_count)[
+            session.bind.dialect.name in ["postgresql", "sqlite", "mysql"] or expected_query_count == 0
+        ]
```

Comment:
```suggestion
        expected_query_count_based_on_db = expected_query_count + 1 if session.bind.dialect.name == ["mssql"] else expected_query_count
```

Reply:
fixed in https://github.com/apache/airflow/pull/9973/commits/6c9df7c658f664c8296af887a2796b3ac082649e